File: API/Models/Purchase.cs
using System;

namespace API.Models
{
    public class Purchase
    {
        public long id { get; set; } = -1;
        public long userID { get; set; } = -1;
        public DateTime orderDate { get; set; } = DateTime.Now;
        public decimal total { get; set; } = -1;
        public string? userName { get; set; }
    }

    public class PurchaseInsert
    {
        public long userID { get; set; } = -1;
        public decimal total { get; set; } = -1;
    }

    public class PurchaseUpdate
    {
        public long id { get; set; } = -1;
        public long userID { get; set; } = -1;
        public decimal total { get; set; } = -1;
    }
}

File: Frontend/src/app/core/models/Purchase.ts
export class Purchase {
  id: number = -1;
  userID: number = -1;
  orderDate: Date = new Date();
  total: number = 0;
  userName?: string;
}

export class PurchaseInsert {
  userID: number = -1;
  total: number = 0;
}

export class PurchaseUpdate {
  id: number = -1;
  userID: number = -1;
  total: number = 0;
}